### PR TITLE
Making GC related fields tunable

### PR DESCRIPTION
**Pull request by diptanu**

### CODE REVIEW

File: command/agent/config-test-fixtures/basic.hcl [hunk 1]
```diff
@@ -53,6 +53,9 @@ client {
         data_points = 35
         collection_interval = "5s"
     }
+    gc_interval = "6s"
+    gc_disk_usage_threshold = 82
+	gc_inode_usage_threshold = 91
```
Comment (schmichael) [inline command/agent/config-test-fixtures/basic.hcl]: Whitespace

File: command/agent/agent.go [hunk 2]
```diff
@@ -288,6 +288,11 @@ func (a *Agent) clientConfig() (*clientconfig.Config, error) {
 	conf.TLSConfig = a.config.TLSConfig
 	conf.Node.TLSEnabled = conf.TLSConfig.EnableHTTP
 
+	// Set the GC related configs
```
Comment (dadgar) [inline command/agent/agent.go]: Can you add the thresholds as flags to the agent

File: client/config/config.go [hunk 3]
```diff
@@ -151,6 +151,18 @@ type Config struct {
 	// TLSConfig holds various TLS related configurations
 	TLSConfig *config.TLSConfig
 
+	// GCInterval is the time interval at which the client triggers garbage
+	// collection
+	GCInterval time.Duration
```
Comment (dadgar) [inline client/config/config.go]: Website docs

**Comment by jippi** [comment]
@diptanu will this allow me to configure nomad to not wipe any allocations before, say, 24h ?

**Comment by dadgar** [comment]
@jippi Nope this just allows picking at what disk usage percentage GC kicks in

**Comment by jippi** [comment]
@dadgar okay, i really want to be able to extend the disk GC to be more than 4h, more than once i've lost template or logs for debugging :)

**Comment by dadgar** [comment]
@jippi https://github.com/hashicorp/nomad/pull/2370

**Pull request merged**